src/Integration/Suppression/SuppressedIssueMatcher.cs

```diff
@@ -40,19 +41,38 @@ public bool SuppressionExists(IFilterableIssue issue)
                 throw new ArgumentNullException(nameof(issue));
             }
 
-            // Issues match if:
+
+            // File-level issues (i.e. line = null) match if:
+            // 1. Same component, same file, same error code.
```

Comment:
I noticed this bug when looking at the Roslyn `LiveIssueFactory`. 

File-level issues have `null` for the `hash` and the line number, and they should not match any non-file-level issues.

This bug existed in the previous code, but no-one noticed/complained.